unified_planning/model/axiom.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2021-2023 AIPlan4EU project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+This module defines the base class `Axiom'.
+An `Axiom' has a `head' which is a single `Parameter' and a `body' which is a single `Condition'.
+"""
+
+from unified_planning.model.action import *
+from unified_planning.environment import get_environment, Environment
+
+
+class Axiom(InstantaneousAction):
```

Comment:
Why deriving from `InstantaneousAction`? This makes it possible to have `Axioms` passed to the `add_action` methods and causes some hacky code below. Why not a dedicated class?

Reply:
Can you have a look at the internal structure of the InstantaneousActiona and see what is that you really need? Maybe this could be a call to a small refactoring aimed at factorising what is needed to both instantaneous actions and axioms.